src/consensus/aft/test/committable_suffix.cpp

```diff
@@ -0,0 +1,860 @@
+    DOCTEST_REQUIRE(1 == dispatch_all(nodes, node_idD, channelsD->messages));
+    DOCTEST_REQUIRE(1 == dispatch_all(nodes, node_idE, channelsE->messages));
+
+    DOCTEST_REQUIRE(rA.is_primary());
+    DOCTEST_REQUIRE(rA.get_term() == 1);
+  }
+
+  DOCTEST_INFO("Entry at 1.1 is received by all nodes");
+  {
+    auto entry = make_ledger_entry(1, 1);
+    rA.replicate(kv::BatchVector{{1, entry, true, hooks}}, 1);
+    DOCTEST_REQUIRE(rA.get_last_idx() == 1);
+    DOCTEST_REQUIRE(rA.get_commit_idx() == 0);
+    // Size limit was reached, so periodic is not needed
+    // rA.periodic(request_timeout);
+
+    // Dispatch AppendEntries
+    DOCTEST_REQUIRE(4 == dispatch_all(nodes, node_idA, channelsA->messages));
+
+    // All nodes have this
+    DOCTEST_REQUIRE(rB.get_last_idx() == 1);
+    DOCTEST_REQUIRE(rC.get_last_idx() == 1);
+    DOCTEST_REQUIRE(rD.get_last_idx() == 1);
+    DOCTEST_REQUIRE(rE.get_last_idx() == 1);
+
+    // Dispatch AppendEntriesResponses
+    DOCTEST_REQUIRE(1 == dispatch_all(nodes, node_idB, channelsB->messages));
+    DOCTEST_REQUIRE(1 == dispatch_all(nodes, node_idC, channelsC->messages));
+    DOCTEST_REQUIRE(1 == dispatch_all(nodes, node_idD, channelsD->messages));
+    DOCTEST_REQUIRE(1 == dispatch_all(nodes, node_idE, channelsE->messages));
+
+    // Node A now knows this is committed
+    DOCTEST_REQUIRE(rA.get_commit_idx() == 1);
+  }
+
+  DOCTEST_INFO(
+    "Entries at 1.2, 1.3, and 1.4 are received by a majority, and become "
+    "committed");
+  {
+    auto entry = make_ledger_entry(1, 2);
+    rA.replicate(kv::BatchVector{{2, entry, true, hooks}}, 1);
+    DOCTEST_REQUIRE(rA.get_last_idx() == 2);
+    DOCTEST_REQUIRE(rA.get_commit_idx() == 1);
+    // Size limit was reached, so periodic is not needed
+    // rA.periodic(request_timeout);
+
+    entry = make_ledger_entry(1, 3);
+    rA.replicate(kv::BatchVector{{3, entry, true, hooks}}, 1);
+    DOCTEST_REQUIRE(rA.get_last_idx() == 3);
+    DOCTEST_REQUIRE(rA.get_commit_idx() == 1);
+    // Size limit was reached, so periodic is not needed
+    // rA.periodic(request_timeout);
+
+    entry = make_ledger_entry(1, 4);
+    rA.replicate(kv::BatchVector{{4, entry, true, hooks}}, 1);
+    DOCTEST_REQUIRE(rA.get_last_idx() == 4);
+    DOCTEST_REQUIRE(rA.get_commit_idx() == 1);
+    // Size limit was reached, so periodic is not needed
+    // rA.periodic(request_timeout);
+
+    {
+      DOCTEST_INFO("Delete the AppendEntries for D and E");
+      auto it = channelsA->messages.begin();
+      while (it != channelsA->messages.end())
+      {
+        auto [to, _] = *it;
+        if (to == node_idD || to == node_idE)
+        {
+          it = channelsA->messages.erase(it);
+        }
+        else
+        {
+          ++it;
+        }
+      }
+    }
+
+    DOCTEST_REQUIRE(6 == dispatch_all(nodes, node_idA, channelsA->messages));
+
+    // NB: AppendEntriesResponses are not dispatched yet. So 1.4 is technically
```

Comment:
I think we ought to replace "technically committed" with "present on f+1 nodes", same with "True commit index", "Highest index present on f+1 nodes" etc, to distinguish that from commit index, which is what's advertised by the primary and eventually acknowledged by replicas.